io/io/src/TKey.cxx

```diff
@@ -1014,9 +1015,13 @@ TObject *TKey::ReadObjWithBuffer(char *bufferRead)
 
 void *TKey::ReadObjectAny(const TClass* expectedClass)
 {
+   if (!expectedClass) {
```

Comment:
This is a new limitation.  It is intended for this routine to have a 'no check mode'.